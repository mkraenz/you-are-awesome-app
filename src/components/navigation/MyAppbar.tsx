import React, { FC } from "react";
import { Appbar } from "react-native-paper";

interface Props {
    title: string;
    icon?: string;
}

const MyAppbar: FC<Props> = ({ title, icon = "menu" }) => {
    return (
        <Appbar.Header>
            <Appbar.Action icon={icon} onPress={() => undefined} />
            <Appbar.Content title={title} />
            <Appbar.Action icon="magnify" onPress={() => undefined} />
            <Appbar.Action icon="dots-vertical" onPress={() => undefined} />
        </Appbar.Header>
    );
};

export default MyAppbar;
