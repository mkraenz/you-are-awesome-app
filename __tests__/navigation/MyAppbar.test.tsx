import React from "react";
import "react-native";
import ShallowRenderer from "react-test-renderer/shallow";
import MyAppbar from "../../src/components/navigation/MyAppbar";

jest.mock("@expo/vector-icons", () => ({ FontAwesome: "Fontawesome" }));

it("renders correctly", () => {
    const renderer = ShallowRenderer.createRenderer();
    renderer.render(<MyAppbar title="my-title" />);
    const tree = renderer.getRenderOutput();

    expect(tree).toMatchSnapshot();
});
