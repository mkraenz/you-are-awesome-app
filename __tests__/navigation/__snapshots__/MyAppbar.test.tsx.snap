// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<withTheme(Appbar.Header)>
  <Appbar.Action
    icon="menu"
    onPress={[Function]}
    size={24}
  />
  <withTheme(Appbar.Content)
    title="my-title"
  />
  <Appbar.Action
    icon="magnify"
    onPress={[Function]}
    size={24}
  />
  <Appbar.Action
    icon="dots-vertical"
    onPress={[Function]}
    size={24}
  />
</withTheme(Appbar.Header)>
`;
